src/adapters/bidderFactory.js

```diff
@@ -296,6 +311,12 @@ export function newBidder(spec) {
               logWarn(`Bidder ${spec.code} made bid for unknown request ID: ${bid.requestId}. Ignoring.`);
             }
           }
+
+          function headerParser(xmlHttpResponse) {
+            return {
+              get: responseObj.getResponseHeader.bind(responseObj)
```

Comment:
why do a bind here?

Reply:
Paranoia... because I've been burned by `this` bugs too many times.

The root of the issue is: I don't know how browsers implement `XMLHttpRequest.getResponseHeader`.

Suppose I just did:

```
return {
  get: responseObj.getResponseHeader
}
```

but then they implemented it as something like this:

```
XmlHttpRequest.prototype.getResponseHeader = function(header) {
  var headers = this.parseResponseHeaders();
  return headers[header];
}
```

When an adapter calls `get`, then `this` evaluates to _our_ object--not the original `XmlHttpRequest`. Since we don't define `parseResponseHeaders`, it'd crash on an undefined function error.

I have no idea if this is a real risk... but the tl;dr is that `this` sucks. Since I can't eject it from the language, defensive binds keep the code safe.

Reply:
ah ok cool